### PR TITLE
if onsite, redirect main prereg form to the at-door kiosk page

### PR DESCRIPTION
**Pull request by binary1230**
still testing, basically this is for attendees visiting the main prereg page, they get directed to the kiosk form which is more at-event friendly and allows buying day passes online.

### CODE REVIEW

File: uber/decorators.py [hunk 1]
```diff
@@ -30,6 +30,15 @@ def with_check(*args, **kwargs):
...
+def redirect_if_at_con_to_kiosk(func):
+    @wraps(func)
+    def with_check(*args, **kwargs):
+        if c.AT_THE_CON:
+            raise HTTPRedirect(c.KIOSK_REDIRECT_URL)
+        return func(*args, **kwargs)
+    return with_check
+
+
 def check_if_can_reg(func):
```
Comment (binary1230) [inline uber/decorators.py]: would actually love to make something like:

```redirect_if(c.AT_THE_CON, c.KIOSK_REDIRECT_URL)```

but don't have time at the moment

**Comment by binary1230** [comment]
works locally fine and looking good.

**Comment by kitsuta** [comment]
+1

File: uber/decorators.py [hunk 2]
```diff
@@ -30,6 +30,15 @@ def with_check(*args, **kwargs):
     return with_check
 
 
+def redirect_if_at_con_to_kiosk(func):
+    @wraps(func)
+    def with_check(*args, **kwargs):
+        if c.AT_THE_CON:
```
Comment (kitsuta) [inline uber/decorators.py]: Minor improvement which I'll probably just go ahead and make now: this should also check for the existence of the URL, so we can 'turn off' this functionality by simply blanking out the kiosk redirect URL

File: uber/decorators.py (hunk 1 above)
Comment (kitsuta) [inline uber/decorators.py]: Y'know, the more I think about this, the more it seems like making it a decorator was a bit too over-engineered. Like, we only use this decorator in one place, and if we made it more flexible, all we'd be doing is making two lines of code into one line.

**Pull request merged**